command/agent/config/config.go

```diff
@@ -236,16 +473,6 @@ func LoadConfig(path string) (*Config, error) {
 		return nil, fmt.Errorf("error parsing 'api_proxy':%w", err)
 	}
 
-	if result.APIProxy != nil && result.Cache != nil {
```

Comment:
These checks have been moved to a new check stage after all of the config has been added and merged.